apps/src/templates/sectionsRefresh/QuickAssignTableHelpers.jsx

```diff
@@ -86,4 +97,24 @@ function updateSectionCourse(updateCourse, course) {
     versionId: courseVersionId,
     unitId: null
   });
+
+  // Determine if it is a stand alone unit
+  const courseVersion = courseVersions[courseVersionId];
+  const isStandaloneUnit = Object.keys(courseVersion.units).length < 2;
+
+  // if so, update the values for LessonExtras and TTS
+  if (isStandaloneUnit) {
+    const hasLessonExtras = Object.values(courseVersion.units)[0]
+      .lesson_extras_available;
+    const hasTextToSpeech = Object.values(courseVersion.units)[0]
+      .text_to_speech_enabled;
+    updateCourse({
+      displayName: course.display_name,
+      courseOfferingId: course.id,
+      versionId: courseVersionId,
+      unitId: null,
+      hasLessonExtras: hasLessonExtras,
```

Comment:
I think you could consolidate the two updateCourse calls by using 'let' for hasLessonExtras and hasTextToSpeech, assigning them if standalone, and setting them in the updateCourse either way (to null if we don't know) for a shorter/cleaner function. Then just double check that the advanced settings usage will ignore them if they exist but are null.